src/webgpu/web_platform/canvas/getCurrentTexture.spec.ts

```diff
@@ -260,3 +265,80 @@ g.test('resize')
     currentTexture = ctx.getCurrentTexture();
     t.expect(prevTexture === currentTexture);
   });
+
+g.test('expiry')
+  .desc(
+    `
+Test automatic WebGPU canvas texture expiry on all canvas types with the following requirements:
+- getCurrentTexture returns the same texture object within the same frame, throughout:
+  - after previous frame update the rendering
+  - before current frame update the rendering
+  - in a microtask off the current frame task
+- getCurrentTexture returns a new texture object and the old texture object becomes invalid as soon as possible after HTML update the rendering.
+  `
+  )
+  .params(u =>
+    u //
+      .combine('canvasType', kAllCanvasTypes)
+      .combine('prevFrameCallsite', ['requestPostAnimationFrame', 'requestAnimationFrame'] as const)
+  )
+  .fn(t => {
+    const { canvasType, prevFrameCallsite } = t.params;
+    const ctx = t.initCanvasContext(t.params.canvasType);
+    // Create a bindGroupLayout to test invalid texture view usage later.
+    const bgl = t.device.createBindGroupLayout({
+      entries: [
+        {
+          binding: 0,
+          visibility: GPUShaderStage.COMPUTE,
+          texture: {},
+        },
+      ],
+    });
+
+    // The fn is called immediately after previous frame updating the rendering.
+    // Polyfill by calling the callback by setTimeout, in the requestAnimationFrame callback (for onscreen canvas)
+    // or after transferToImageBitmap (for offscreen canvas).
+    function requestPostAnimationFrame(fn: () => void) {
+      t.requestNewFrameOnCanvasType(canvasType, ctx, () => {
+        timeout(fn);
+      });
+    }
+
+    function checkGetCurrentTexture() {
+      // Call getCurrentTexture on previous frame.
+      const prevTexture = ctx.getCurrentTexture();
+
+      // Call getCurrentTexture immediately after the frame, the texture object should stay the same.
+      queueMicrotask(() => {
+        t.expect(prevTexture === ctx.getCurrentTexture());
+      });
+
+      // Call getCurrentTexture immediately after this frame updating the rendering.
+      // It should return a new texture object.
+      timeout(() => {
+        t.expect(prevTexture !== ctx.getCurrentTexture());
+
```

Comment:
Also, not totally sure - but @kainino0x should know.

We should have a tests that don't use rAF for:

```
timeout(() => {
  const t = ctx.getCurrentTexture();
  timeout(() => {
    // is using `t` valid???
  });
});
```

```
timeout(() => {
  const t = ctx.getCurrentTexture();
  timeout(() => {
    const t2 = ctx.getCurrentTexture();
     // does t === t2??
  });
});
```